KSP codegen fails on data class w/ interface

given a class

```kotlin
interface SomeInterface{
	val a : String
}

data class SomeImpl(override val a : String, val b : String) : SomeInterface
```

I get the following error: 

```
> Task :common:kaptGenerateStubsKotlin FAILED

...

e: [ksp] supertype [...].SomeInterface is not a Kotlin type

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':common:kaptGenerateStubsKotlin'.
> Compilation error. See log for more details
```

(package and class name modified to make this example easier to read)

As an aside, is that `kapt` task name a red herring?

--- a/moshi-ksp/moshi-ksp/src/main/kotlin/dev/zacsweers/moshix/ksp/KspUtil.kt
+++ b/moshi-ksp/moshi-ksp/src/main/kotlin/dev/zacsweers/moshix/ksp/KspUtil.kt
@@ -1,7 +1,10 @@
 package dev.zacsweers.moshix.ksp
 
+import com.google.devtools.ksp.getAllSuperTypes
+import com.google.devtools.ksp.processing.KSBuiltIns
 import com.google.devtools.ksp.processing.KSPLogger
 import com.google.devtools.ksp.processing.Resolver
+import com.google.devtools.ksp.symbol.ClassKind.CLASS
 import com.google.devtools.ksp.symbol.KSAnnotated
 import com.google.devtools.ksp.symbol.KSAnnotation
 import com.google.devtools.ksp.symbol.KSClassDeclaration
@@ -30,6 +33,13 @@ internal fun Resolver.getClassDeclarationByName(fqcn: String): KSClassDeclaratio
 }
 
 internal fun KSClassDeclaration.asType() = asType(emptyList())
+
+internal fun KSClassDeclaration.superclass(resolver: Resolver): KSType {
+  return getAllSuperTypes().firstOrNull {
+    val decl = it.declaration
+    decl is KSClassDeclaration && decl.classKind == CLASS
+  } ?: resolver.builtIns.anyType
+}
 
 internal fun KSAnnotated.hasAnnotation(target: KSType): Boolean {
   return findAnnotationWithType(target) != null

--- a/moshi-ksp/moshi-ksp/src/main/kotlin/dev/zacsweers/moshix/ksp/TargetTypes.kt
+++ b/moshi-ksp/moshi-ksp/src/main/kotlin/dev/zacsweers/moshix/ksp/TargetTypes.kt
@@ -1,6 +1,5 @@
 package dev.zacsweers.moshix.ksp
 
-import com.google.devtools.ksp.getAllSuperTypes
 import com.google.devtools.ksp.getDeclaredProperties
 import com.google.devtools.ksp.getVisibility
 import com.google.devtools.ksp.isInternal
@@ -9,6 +8,7 @@ import com.google.devtools.ksp.isPublic
 import com.google.devtools.ksp.processing.KSPLogger
 import com.google.devtools.ksp.processing.Resolver
 import com.google.devtools.ksp.symbol.ClassKind
+import com.google.devtools.ksp.symbol.ClassKind.CLASS
 import com.google.devtools.ksp.symbol.KSAnnotated
 import com.google.devtools.ksp.symbol.KSClassDeclaration
 import com.google.devtools.ksp.symbol.KSDeclaration
@@ -39,7 +39,7 @@ internal fun targetType(
   logger.check(type.classKind != ClassKind.ENUM_CLASS, type) {
     "@JsonClass with 'generateAdapter = \"true\"' can't be applied to ${type.qualifiedName?.asString()}: code gen for enums is not supported or necessary"
   }
-  logger.check(type.classKind == ClassKind.CLASS, type) {
+  logger.check(type.classKind == CLASS, type) {
     "@JsonClass can't be applied to ${type.qualifiedName?.asString()}: must be a Kotlin class"
   }
   logger.check(Modifier.INNER !in type.modifiers, type) {
@@ -76,7 +76,7 @@ internal fun targetType(
   val superTypes = appliedType.supertypes(resolver, logger)
     .filterNot { supertype ->
       supertype.typeName == OBJECT_CLASS || // Don't load properties for java.lang.Object.
-        supertype.type.classKind != ClassKind.CLASS  // Don't load properties for interface types.
+        supertype.type.classKind != CLASS  // Don't load properties for interface types.
     }
     .onEach { supertype ->
       if (supertype.type.hasAnnotation(resolver.getClassDeclarationByName<Metadata>().asType())) {
@@ -89,8 +89,8 @@ internal fun targetType(
       // Load the kotlin API cache into memory eagerly so we can reuse the parsed APIs
       val api = supertype.type
 
-      val apiSuperClass = api.getAllSuperTypes().firstOrNull()
-      if (apiSuperClass != null && apiSuperClass.arguments.isNotEmpty()) {
+      val apiSuperClass = api.superclass(resolver)
+      if (apiSuperClass != resolver.builtIns.anyType && apiSuperClass.arguments.isNotEmpty()) {
         //
         // This extends a typed generic superclass. We want to construct a mapping of the
         // superclass typevar names to their materialized types here.

--- a/moshi-ksp/tests/src/test/kotlin/dev/zacsweers/moshix/ksp/codegen/MoshiKspTest.kt
+++ b/moshi-ksp/tests/src/test/kotlin/dev/zacsweers/moshix/ksp/codegen/MoshiKspTest.kt
@@ -1,0 +1,33 @@
+package dev.zacsweers.moshix.ksp.codegen
+
+import com.google.common.truth.Truth.assertThat
+import com.squareup.moshi.JsonClass
+import com.squareup.moshi.Moshi
+import dev.zacsweers.moshix.adapter
+import org.junit.Test
+
+// Tests specific to Moshi-KSP
+class MoshiKspTest {
+  private val moshi = Moshi.Builder().build()
+
+  // Regression test for https://github.com/ZacSweers/MoshiX/issues/44
+  @Test
+  fun onlyInterfaceSupertypes() {
+    val adapter = moshi.adapter<SimpleImpl>()
+    //language=JSON
+    val json = """{"a":"aValue","b":"bValue"}"""
+    val expected = SimpleImpl("aValue", "bValue")
+    val instance = adapter.fromJson(json)!!
+    assertThat(instance).isEqualTo(expected)
+    val encoded = adapter.toJson(instance)
+    assertThat(encoded).isEqualTo(json)
+  }
+
+  interface SimpleInterface{
+    val a : String
+  }
+
+  @JsonClass(generateAdapter = true)
+  data class SimpleImpl(override val a : String, val b : String) : SimpleInterface
+}
+

